cmd/ike/session/session.go

```diff
@@ -143,26 +135,26 @@ func waitForRefToComplete(sessionName, ref string) (string, error) {
 	return name, nil
 }
 
-func removeOrLeaveSession(sessionName, ref string) {
-	session, err := DefaultClient().Get(sessionName)
+func (h *handler) removeOrLeaveSession() {
+	session, err := h.c.Get(h.sessionName)
 	if err != nil {
 		return // assume missing, nothing to clean?
 	}
-	// more then one participant, update session
+	// more then one participant, update sessionName
 	for i, r := range session.Spec.Refs {
-		if r == ref {
+		if r == h.ref {
 			session.Spec.Refs = append(session.Spec.Refs[:i], session.Spec.Refs[i+1:]...)
 		}
 	}
 	if len(session.Spec.Refs) == 0 {
-		_ = DefaultClient().Delete(session)
+		_ = h.c.Delete(session)
 	} else {
-		_ = DefaultClient().Create(session)
+		_ = h.c.Create(session)
 	}
 }
 
 func getSessionName(cmd *cobra.Command) string {
-	sessionName, err := cmd.Flags().GetString("session")
+	sessionName, err := cmd.Flags().GetString("sessionName")
```

Comment:
That this passes our test suite is a little scary: https://github.com/aslakknutsen/istio-workspace/blob/de8a5f7c4c09e8e98349ab1d59e72b52ec3e9427/cmd/ike/cmd/develop.go#L79

Reply:
Lemme have a look :) but yeah, it is

Reply:
```suggestion
	sessionName, err := cmd.Flags().GetString("session")
```

Reply:
It seems we don't cover a case when we want to join an existing session - so even with this regression we anyway always created a new one. 

Joining existing would be a bit tricky with starting 2 TP instances, but I can provide a test where I create Session CR upfront and join it.

Reply:
Hmm, shouldn't we technically end in a "err" here? https://github.com/aslakknutsen/istio-workspace/blob/36cdc4daddd549aa9577dac27cc253d77de8dc24/cmd/ike/session/session.go#L158 

Or is that only on type parse error?

Reply:
Yeah, that too. I already changed that.

About joining an existing session - maybe we could provide an option to persist a session, as currently when we stop `ike develop` we always clean up. How about opening an issue for that and ship the test together with it?

Reply:
Hmm.. That would leave a Route in error when we tear down the Local service it points to. We could for the sake of test, or in prep for a future feature of 'the monitoring option/build image' replacement type of thing... 

Maybe just leave it as to why we need this? https://github.com/aslakknutsen/istio-workspace/issues/59 :)

Reply:
> That would leave a Route in error when we tear down the Local service it points to.

Correct - we would need to check if there are any "active" participants first, before routing. This also opens up a question on how to handle two participants working on the same service anyway (from the session concept point of view). Something we should discuss in depth I believe

Reply:
I added logging, as this is parse error most likely. In such a case, I think we should roll back to random session name.